PICMI_Python/base.py

```diff
@@ -51,3 +52,30 @@ def init(self, kw):
         # --- The implementation of this routine should use kw.pop() to retrieve input arguments from kw.
         # --- This allows testing for any unused arguments and raising an error if found.
         pass
+
+    def check_unsupported_argument(self, arg_name, extra_info=None, raise_error=False):
+        """Raise a warning or exception for an unsupported argument."""
+        # --- Implementation note: This should be called in the "init" method of the implementing
+        # --- class for each unsupported argument. For example, for the 'density_scale' argument of Species:
+        # ---     self.check_unsupported_argument('density_scale', extra_info='My code can not handle a density_scale')
+        if getattr(self, arg_name) is not None:
```

Comment:
Do we check against `None` b/c this is the value if *the user does not touch anything* or because it signifies explicitly *I do not want this feature*?

-> Do we need a `default_value` as additional param, which defaults to `None` if not given?

Reply:
Most arguments to the classes have None as their default values. It is the former - a value that is not None means that the argument was specified by the user.

I would be hesitant to add `default_value` as an additional argument since that would need to have the default value duplicated in the call and this is not robust to code changes.